contracts/data/IncrementalMerkleTree2.sol

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import 'hardhat/console.sol';
+
+library IncrementalMerkleTree2 {
+    using IncrementalMerkleTree2 for Tree;
+
+    struct Tree {
+        bytes32[][] nodes;
```

Comment:
@NouDaimon I think the main disadvantage of this approach is the lack of support an `indexOf` function.  However, I don't think it's a fundamental part of the data structure, and end users could add their own implementations if they want to spend the gas.